keps/sig-node/2400-node-swap/README.md

```diff
@@ -0,0 +1,632 @@
+- Clusters with nodes \<4Gi memory https://github.com/kubernetes/kubernetes/issues/53533#issuecomment-751404417 
+
+This user story is addressed by scenario 2, and could benefit from 3.
+
+#### Virtualization management overhead
+
+This would apply to virtualized Kubernetes workloads such as VMs launched by kubevirt.
+
+Every VM comes with a management related overhead which can sporadically be pretty significant (memory streaming, SRIOV attachment, gpu attachment, virtio-fs, …). Swap helps to not request much more memory to deal with short term worst-case scenarios.
+
+With virtualization, clusters are typically provisioned based on the workloads’ memory consumption, and any infrastructure container overhead is overcommitted. This overhead could be safely swapped out.
+
+- Required for live migration of VMs https://github.com/kubernetes/kubernetes/issues/53533#issuecomment-754878431 
+
+This user story is addressed by scenario 2, and could benefit from 3.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+Having swap available on a system reduces predictability. When swap is available to workloads, and is not accounted for on an individual workload-by-workload basis
+
+First, this risk is mitigated by preventing any workloads from using swap by default, even if it is enabled on a system. This will allow a cluster administrator to test swap utilization just at the system level without introducing unpredictability to workload resource utilization.
+
+Additionally, we mitigate this risk by quantifying system stability and then gathering test and production data to determine if system stability remains the same or is improved when swap is available to the system and/or workloads.
+
+Since swap provisioning is out of scope of this proposal, this enhancement poses little risk to Kubernetes clusters that will not enable swap.
+
+## Design Details
+
+\[In progress\]
+
+Need to add specifics here for:
+
+- Changes to `--fail-on-swap` flag
+- CRI config details
+- Where changes will need to be made so that dockershim and the CRI are consistent with swap control
+
+### Test Plan
+
+For alpha:
+
+- Swap scenarios are enabled in test-infra for at least two Linux distributions. e2e suites will be run against them.
+- Data should be gathered from a number of use cases to guide beta graduation and further development efforts.
+
+Once this data is available, additional test plans should be added for the next phase of graduation.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Kubelet can be started with swap enabled.
+- KubeletConfig allows CRI to be configured with a percentage of swap available to workloads. This will default to 0.
+- e2e test jobs are configured for Linux systems with swap enabled.
```

Comment:
add a test that validates that the eviction thresholds only account for memory in RAM and when memory was offloaded to swap, it is not accounted and not lead to eviction.

Reply:
Added above to alpha test plan.